### ARCHIVED (not applied by default builds)
#
# This patch is kept around in case we later need to support topologies where
# multiple clients appear from the same IP (different source ports), or where
# BAN_TEST's POSIX fallback struct definitions cause trouble on some platforms.
#
--- net_dgrm.c
+++ net_dgrm.c
@@ -29,26 +29,9 @@
 #include <sys/socket.h>
 #include <arpa/inet.h>
 #else
-#define AF_INET 		2	/* internet */
-struct in_addr
-{
-	union
-	{
-		struct { unsigned char s_b1,s_b2,s_b3,s_b4; } S_un_b;
-		struct { unsigned short s_w1,s_w2; } S_un_w;
-		unsigned long S_addr;
-	} S_un;
-};
-#define	s_addr	S_un.S_addr	/* can be used for most tcp & ip code */
-struct sockaddr_in
-{
-    short			sin_family;
-    unsigned short	sin_port;
-	struct in_addr	sin_addr;
-    char			sin_zero[8];
-};
-char *inet_ntoa(struct in_addr in);
-unsigned long inet_addr(const char *cp);
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
 #endif
 #endif	// BAN_TEST

@@ -994,7 +977,7 @@
 		if (s->driver != net_driverlevel)
 			continue;
 		ret = dfunc.AddrCompare(&clientaddr, &s->addr);
-		if (ret >= 0)
+		if (ret == 0)
 		{
 			// is this a duplicate connection reqeust?
 			if (ret == 0 && net_time - s->connecttime < 2.0)
 			{
